packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -377,10 +383,20 @@ class ScaffoldMessengerState extends State<ScaffoldMessenger> with TickerProvide
 
   void _updateScaffolds() {
     for (final ScaffoldState scaffold in _scaffolds) {
-      scaffold._updateSnackBar();
+      if (_isRoot(scaffold))
+        scaffold._updateSnackBar();
```

Comment:
At this point it would be safe to return.

Reply:
Not necessarily. There can be other Scaffolds that need to get a SnackBar. For example, on another route. That is how we get the nice hero animation as the SnackBar persists across the transition.

Reply:
I see. Hopefully there's a test that will catch someone (like me) adding the short-circuit return without realizing the implications